### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by rymurr**
left todo:

* implement history and forward/backward navigation of tables listing
* ensure non-iceberg tables look ok
* add testing!
* delete unused code in `_old`
* reenable create/merge of branches

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/369)

### CODE REVIEW

File: ui/src/App.css [hunk 1]
```diff
@@ -13,9 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-.App {
-  text-align: center;
-}
+/*.App {*/
```
Comment (thwong) [inline ui/src/App.css:16]: Remove comment.

File: ui/src/App.css [hunk 2]
```diff
@@ -51,3 +51,8 @@
     transform: rotate(360deg);
   }
 }
+
+.modal-90w {
+  width: 90%;
+  max-width: none!important;
```
Comment (thwong) [inline ui/src/App.css:57]: Space before "!important"

File: ui/src/App.js [hunk 3]
```diff
@@ -20,11 +20,12 @@ import {Router, Route} from 'react-router-dom';
 import {history} from './utils';
```
Comment (thwong) [inline ui/src/App.js:20]: Remove commented out codes in this file.

File: ui/src/ContentsModal/ContentsModal.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  name.push(props.currentKey.name);
+  const [currentContent, setContent] = useState([]);
+  useEffect(() => {
+    getContents(name, props.currentBranch, setContent);
+  }, [props.currentKey])
+
+  return (
+    <Modal show={props.show} onHide={props.handleClose} dialogClassName="modal-90w" size="large">
+      <Modal.Header closeButton>
+        <Modal.Title>{name.join(".") + " on " + props.currentBranch.name} </Modal.Title>
```
Comment (thwong) [inline ui/src/ContentsModal/ContentsModal.jsx:85]: For string interpolation in JS, use ` `
i.e <Modal.Title>{`${name.join("."} on ${props.currentBranch.name}`}

File: ui/src/utils/api.js [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        const value = obj[key];
+        if (value === undefined) {
+          return '';
+        }
+        if (value === null) {
+          return key;
+        }
+        if (Array.isArray(value)) {
+          if (value.length) {
+            return key + '=' + value.map(encodeURIComponent).join('&' + key + '=');
```
Comment (thwong) [inline ui/src/utils/api.js:20]: See the string interpolation example from above ... using ``

File: ui/src/utils/api.js [hunk 6]
```diff
@@ -0,0 +1,353 @@
+export default createApi;
```
Comment (thwong) [inline ui/src/utils/api.js:1]: Might want to consider breaking up these API calls into separate files so the API calls are organized better..
 
/utils/api/
 - index.js
 - reference.js
 - content.js
 - branch.js 
 
etc ...

Comment (rymurr) [reply ui/src/utils/api.js:17]: Hey Tim, thanks for the review. This file is auto-generated by a script. Hence it being all in one messy file. The process hasn't been automated yet but will be in a follow up PR

**Review by rymurr — COMMENTED** [review]
> Reviewed 10 of 34 files at r1, 13 of 18 files at r2, 25 of 27 files at r3, 15 of 28 files at r5.
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/369)* status: 36 of 48 files reviewed, 15 unresolved discussions (waiting on @rymurr and @thwong)

___
*[ui/src/App.js, line 24 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkXsluEe47m7MyLpSy:-MKkXsluEe47m7MyLpSz:b-pannz8) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/App.js#L24)):*
> ```JavaScript
> 
>       </Navbar>
>       <Switch>
> ```

quick check shows that `Switch` is a helper of some sort to react `Router`? Is that your understanding/motivation to switch to it?
___
*[ui/src/Explore/Explore.jsx, line 78 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_iIf-NDf4fv5Ohc7:-MKk_iIf-NDf4fv5Ohc8:b-l6mwxj) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L78)):*
> ```JSX
>     })
>     .then((data) => {
>       //setKeys(data.)
> ```

remove
___
*[ui/src/Explore/Explore.jsx, line 137 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_zWsBTBashatGHm3:-MKk_zWt7K3CeDJcPo_Z:b-udku3m) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L137)):*
> ```JSX
> }
> 
> async function updateState(currentUser, slug, setDefaultBranch, setBranches, setTags, setPath, setKeys, setCurrentRef, setLog) {
> ```

to me we are hand managing async state when we could have most of these as state of the component and  let react figure everything out
___
*[ui/src/Explore/Explore.jsx, line 140 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_jhg2IqimpqAD2GY:-MKk_jhg2IqimpqAD2GZ:b-l6mwxj) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L140)):*
> ```JSX
>   const defaultBranch = await fetchDefaultBranch(currentUser);
>   const branchesAndTags = await loadBranchesAndTags();
>   //const [defaultBranch, branchesAndTags] = Promise.all([t => fetchDefaultBranch(currentUser), loadBranchesAndTags]);
> ```

remove
___
*[ui/src/Explore/Explore.jsx, line 158 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkaOi8AcsuRre9tYoA:-MKkaOi8AcsuRre9tYoB:b-l6mwxj) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L158)):*
> ```JSX
>   let history = useHistory();
>   const currentUser = authenticationService.currentUserValue;
>   //const [currentRef, setCurrentRef] = useState("main");
> ```

remove
___
*[ui/src/Explore/TableHead.jsx, line 25 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkam7rEbGn0nn_0hzU:-MKkam7rEbGn0nn_0hzV:b-l6mwxj) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/TableHead.jsx#L25)):*
> ```JSX
> function TableHead(props) {
>   let additional;
>   console.log("TableHead", props);
> ```

remove
___
*[ui/src/Explore/TableHead.jsx, line 39 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkbLuiBBU30tKmXzmS:-MKkbLuiBBU30tKmXzmT:b-at7ae) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/TableHead.jsx#L39)):*
> ```JSX
>         {
>           props.path.map((p, index) => {
>             return (<Fragment><span style={{"padding-left":7, "padding-right":7}}>/</span><ExploreLink currentRef={props.currentRef} path={props.path.slice(0,index+1)}>{p}</ExploreLink></Fragment>);
> ```

this looks weird to me. 7 what (px etc)? I prefer the bootstrap padding classes as they are reactive and well defined.
___
*[ui/src/Explore/TableHead.jsx, line 63 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkb7No4EL0qhZRr6lz:-MKkb7No4EL0qhZRr6m-:b-l6mwxj) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/TableHead.jsx#L63)):*
> ```JSX
>           return (<NavDropdown.Item as={"button"} key={tag.name}><ExploreLink currentRef={tag} path={props.path} type="CONTAINER">{tag}</ExploreLink></NavDropdown.Item>)
>         })}
>         {/*<NavDropdown.Divider/>*/}
> ```

remove
___
*[ui/src/Explore/TableListing.jsx, line 62 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkbr7y-SuRa-YhG5cN:-MKkbr7y-SuRa-YhG5cO:b-hoz6xl) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/TableListing.jsx#L62)):*
> ```JSX
>       </Card>
>       {/*<ContentsModal show={contentsModalState} handleClose={() => setContentsModalState(false)}*/}
>       {/*               currentUser={currentUser} currentBranch={currentBranch} currentKey={contentsSelected}/>*/}
> ```

more commented out code

**Comment by rymurr** [comment]
looks good to me. There is still a lot of dead code and commented lines/console logging. Once that and Tims comments are addressed I am mostly happy. 

I think the only design point worth addressing is the async state update. To me it feels like react should do it instead of us. This will also help us not have to update everythign on every API call

File: ui/public/index.html [hunk 7]
```diff
@@ -17,51 +17,51 @@
...
-      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
-    -->
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+<head>
+  <meta charset="utf-8"/>
+  <link href="%PUBLIC_URL%/favicon.ico" rel="icon"/>
+  <meta content="width=device-width, initial-scale=1" name="viewport"/>
+  <meta content="#000000" name="theme-color"/>
+  <meta
+    content="Web site created using create-react-app"
```
Comment (thwong) [inline ui/public/index.html:26]: Should update the description :-) ... especially if this is a public site that is meant to be crawlable by search engine.

File: ui/src/ContentsModal/ContentsModal.jsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, {useState, useEffect} from 'react';
+
+import {Row,Col} from "react-bootstrap";
```
Comment (thwong) [inline ui/src/ContentsModal/ContentsModal.jsx:18]: ... super minor.  Missing space after ","

File: ui/src/ContentsModal/ContentsModal.jsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    })
+    .then((data) => {
+      setContent(data);
+    })
+    .catch(console.log);
+}
+
+function image(currentContent) {
+  if (currentContent.type === "ICEBERG_TABLE") {
+    return (<Row><Col xs={3}>Table type:</Col> <Col>Iceberg Table<img className="pl-2" src={"/iceberg.png"}
```
Comment (thwong) [inline ui/src/ContentsModal/ContentsModal.jsx:40]: Spacing looks very off... better reformat it ... something like this.

```
return (
     <Row>
          <Col xs={3}>Table type:</Col>
          <Col>
             Iceberg Table
             <img
                  className="pl-2"
                  src={"/iceberg.png"}
                  alt={"iceberg table"}
                  height={"50"} 
                  width={"60"}
             />
         </Col>
    </Row>
)
```

File: ui/src/Explore/Explore.jsx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+import {Card, Container} from "react-bootstrap";
+import {useParams} from "react-router-dom";
+import TableHead from "./TableHead";
+import TableListing from "./TableListing";
+import CommitLog from "./CommitHeader";
+import {api} from "../utils";
+
+
+function fetchDefaultBranch(setDefaultBranch) {
+  //const currentUser = authenticationService.currentUserValue;
```
Comment (thwong) [inline ui/src/Explore/Explore.jsx:26]: Note needed... remove.

File: ui/src/Explore/Explore.jsx [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+    <div>
+      <Container style={{"marginTop": "100px"}}>
+        <Card>
+          <TableHead
+            branches={branches}
+            tags={tags}
+            currentRef={currentRef}
+            defaultBranch={defaultBranch}
+            path={path}
+             />
```
Comment (thwong) [inline ui/src/Explore/Explore.jsx:128]: Minor.  Alignment is a bit off on the closing "/>" ... should be left-indent

File: ui/src/Explore/TableListing.jsx [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  )
+}
+
+function fetchKeys(ref, path, setKeys) {
+  return api().getEntries({'ref':ref})
+    .then(res => {
+      return res.json();
+    })
+    .then((data) => {
+      //setKeys(data.)
```
Comment (thwong) [inline ui/src/Explore/TableListing.jsx:42]: Remove comment

File: ui/src/services/index.js [hunk 13]
```diff
@@ -15,4 +15,4 @@
  */
 export * from './authentication.service';
 export * from './user.service';
-export * from './CreateBranch';
+// export * from './CreateBranch';
```
Comment (thwong) [inline ui/src/services/index.js:18]: Remove comment

**Review by thwong — DISMISSED** [review]
Looks great 👍  ... have some very minor cleanup comments.

**Comment by jacques-n** [comment]
Thanks @thwong, addressed your comments.

**Review by jacques-n — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/369)* status: 18 of 37 files reviewed, 22 unresolved discussions (waiting on @rymurr and @thwong)

___
*[ui/src/App.css, line 16 at r1](https://reviewable.io/reviews/projectnessie/nessie/369#-MK5tsptoD4RIQCAkIFb-r1-16:-MKleZZW0BoiIKbNXPcK:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/47a26f73d258f3a07893df9225cebc068ff753d8/ui/src/App.css#L16)):*
<details><summary><i>Previously, thwong (Tim Wong) wrote…</i></summary><blockquote>

Remove comment.
</blockquote></details>

done
___
*[ui/src/App.css, line 57 at r1](https://reviewable.io/reviews/projectnessie/nessie/369#-MK5tsptoD4RIQCAkIFb-r1-57:-MKle_BL299DRGgXK93k:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/47a26f73d258f3a07893df9225cebc068ff753d8/ui/src/App.css#L57)):*
<details><summary><i>Previously, thwong (Tim Wong) wrote…</i></summary><blockquote>

Space before "!important"
</blockquote></details>

done
___
*[ui/src/App.js, line 20 at r1](https://reviewable.io/reviews/projectnessie/nessie/369#-MK5tsptoD4RIQCAkIFc-r1-20:-MKledSV9LlbfENNyVNz:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/47a26f73d258f3a07893df9225cebc068ff753d8/ui/src/App.js#L20)):*
<details><summary><i>Previously, thwong (Tim Wong) wrote…</i></summary><blockquote>

Remove commented out codes in this file.
</blockquote></details>

done
___
*[ui/src/App.js, line 24 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkXsluEe47m7MyLpSy:-MKleegT4Jui-R9gK7eC:bd6e9st) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/App.js#L24)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

quick check shows that `Switch` is a helper of some sort to react `Router`? Is that your understanding/motivation to switch to it?
</blockquote></details>

Yes. It seems like the way to do routing (based on the tutorials I've read).
___
*[ui/src/ContentsModal/ContentsModal.jsx, line 85 at r1](https://reviewable.io/reviews/projectnessie/nessie/369#-MK5tsptoD4RIQCAkIFy-r1-85:-MKlhW4h9IhpX_JtXdFi:b-i4hi66) ([raw file](https://github.com/projectnessie/nessie/blob/47a26f73d258f3a07893df9225cebc068ff753d8/ui/src/ContentsModal/ContentsModal.jsx#L85)):*
<details><summary><i>Previously, thwong (Tim Wong) wrote…</i></summary><blockquote>

For string interpolation in JS, use ` `
i.e <Modal.Title>{`${name.join("."} on ${props.currentBranch.name}`}
</blockquote></details>

I think you meant { `${name.join(".")} on ${props.currentBranch.name} `}

Oh, I bet you accidentally encoded your answer (seeing the gray in your comment now).
___
*[ui/src/Explore/Explore.jsx, line 78 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_iIf-NDf4fv5Ohc7:-MKlo-1Q4uRcxOgvClQP:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L78)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

remove
</blockquote></details>

done
___
*[ui/src/Explore/Explore.jsx, line 137 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_zWsBTBashatGHm3:-MKlnz5T0nfGG16dSKJ-:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L137)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

to me we are hand managing async state when we could have most of these as state of the component and  let react figure everything out
</blockquote></details>

done
___
*[ui/src/Explore/Explore.jsx, line 140 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKk_jhg2IqimpqAD2GY:-MKlnzZv-7BPj1Cg4Tqd:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L140)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

remove
</blockquote></details>

done
___
*[ui/src/Explore/Explore.jsx, line 158 at r3](https://reviewable.io/reviews/projectnessie/nessie/369#-MKkaOi8AcsuRre9tYoA:-MKlnyPx2Hqk4BcuW_mW:bb7f2mf) ([raw file](https://github.com/projectnessie/nessie/blob/8d7286a8bf878488d692eb69e330d16477c23b90/ui/src/Explore/Explore.jsx#L158)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

remove
</blockquote></details>

done

**Review by rymurr — COMMENTED** [review]
> Reviewed 15 of 31 files at r7, 4 of 4 files at r8.
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/369)* status: all files reviewed, 23 unresolved discussions (waiting on @rymurr and @thwong)

___
*[ui/src/Explore/CommitHeader.jsx, line 52 at r8](https://reviewable.io/reviews/projectnessie/nessie/369#-MKpOjIL2TsLO8sX8Kxv:-MKpOjIM30L2hQ_sZLup:b-f2kwia) ([raw file](https://github.com/projectnessie/nessie/blob/8993111e56602871e5ee508e0aacac3f7070bf86/ui/src/Explore/CommitHeader.jsx#L52)):*
> ```JSX
>     </span>
>     <span className={"float-right"}>
>       <span className="font-italic">{currentLog.hash.slice(0,8)}</span>
> ```

we dont have any facility to 'expand' a hash like git does. Its ugly visual noise to leave the entire hash in but is hard to trace back if we slice it

**Review by jacques-n — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/369)* status: all files reviewed, 23 unresolved discussions (waiting on @rymurr and @thwong)

___
*[ui/src/Explore/CommitHeader.jsx, line 52 at r8](https://reviewable.io/reviews/projectnessie/nessie/369#-MKpOjIL2TsLO8sX8Kxv:-MKpQpTD6OoviqNZC3lZ:b-mk0tf9) ([raw file](https://github.com/projectnessie/nessie/blob/8993111e56602871e5ee508e0aacac3f7070bf86/ui/src/Explore/CommitHeader.jsx#L52)):*
<details><summary><i>Previously, rymurr (Ryan Murray) wrote…</i></summary><blockquote>

we dont have any facility to 'expand' a hash like git does. Its ugly visual noise to leave the entire hash in but is hard to trace back if we slice it
</blockquote></details>

Once we link this, I plan to show the full hash on the destination page.

**Review by jacques-n — APPROVED** [review]
Combined approval with @rymurr

**Pull request merged**